data_model/src/block.rs

```diff
@@ -6,347 +6,338 @@
 
 #[cfg(not(feature = "std"))]
 use alloc::{boxed::Box, format, string::String, vec::Vec};
-use core::{cmp::Ordering, fmt::Display};
+use core::{cmp::Ordering, fmt::Display, time::Duration};
 
 use derive_more::Display;
 use getset::Getters;
-#[cfg(feature = "std")]
-use iroha_crypto::SignatureOf;
-use iroha_crypto::{Hash, HashOf, MerkleTree, SignaturesOf};
+use iroha_crypto::{HashOf, KeyPair, MerkleTree, SignaturesOf};
+use iroha_data_model_derive::model;
+use iroha_macro::FromVariant;
 use iroha_schema::IntoSchema;
-use iroha_version::{declare_versioned_with_scale, version_with_scale};
+use iroha_version::{declare_versioned, version_with_scale};
 use parity_scale_codec::{Decode, Encode};
 use serde::{Deserialize, Serialize};
 
-pub use self::{
-    committed::{CommittedBlock, PartialBlockHash, VersionedCommittedBlock},
-    header::BlockHeader,
-};
-use crate::{events::prelude::*, model, peer, transaction::prelude::*};
+pub use self::model::*;
+use crate::{events::prelude::*, peer, transaction::prelude::*};
 
-mod header {
-    pub use self::model::*;
+#[model]
+pub mod model {
     use super::*;
 
-    #[model]
-    pub mod model {
-        use super::*;
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Getters,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    #[cfg_attr(
+        feature = "std",
+        display(fmt = "Block №{height} (hash: {});", "HashOf::new(&self)")
+    )]
+    #[cfg_attr(not(feature = "std"), display(fmt = "Block №{height}"))]
+    #[getset(get = "pub")]
+    #[allow(missing_docs)]
+    #[ffi_type]
+    // TODO: Do we need both BlockPayload and BlockHeader?
+    // If yes, what data goes into which structure?
```

Comment:
yes, initially we had only `BlockHeader` but now we also have `BlockPayload` because of `hash` to return what was previously called `PartialHash`

we don't implement `Hash` anymore and both `Eq` and `Ord` will short-circuit so there is no cost to deriving them

Reply:
opened #3895 for further discussion